osc_counter/oscserver.py

```diff
@@ -4,31 +4,49 @@
 from filemanager import FileManager
 from timechecker import TimeChecker
 from messenger import Messenger
+from usersettings import UserSettings
+import ctypes
+import time
 import threading
+import datetime
+import logging
 
 class OSCServer():
     def __init__(self, filemanager, messenger):
+        self.user_settings: UserSettings = UserSettings()
         self.dispatcher = Dispatcher()
-        self.dispatcher.set_default_handler(self._def_osc_dispatch)
-        self.server = BlockingOSCUDPServer(("127.0.0.1", 9001), self.dispatcher)
-        self.server_thread = threading.Thread(target=self._process_osc)
-        self.client =  udp_client.SimpleUDPClient("127.0.0.1", 9000)
         self.filemanager: FileManager = filemanager
         self.messenger: Messenger = messenger
-
+        self.dispatcher.set_default_handler(self._def_osc_dispatch)
+        logging.basicConfig(filename='output.txt', level=logging.INFO, format='')
+        try:
+            self.server = BlockingOSCUDPServer((self.user_settings.IP, self.user_settings.ListeningPort), self.dispatcher)
+            self.server_thread = threading.Thread(target=self._process_osc)
+            self.client =  udp_client.SimpleUDPClient(self.user_settings.IP, self.user_settings.SendingPort)
+        except Exception as e: 
+            print(e)
+            logging.error(e)
+            time.sleep(5)
+        print(f"IP: {self.user_settings.IP}")
+        print(f"Listening on port: {self.user_settings.ListeningPort}\nSending on port: {self.user_settings.SendingPort}")
+  
     def launch(self) -> None:
         self.server_thread.start()
+        print("VRC Contact Counter is Running!") 
 
     def shutdown(self) -> None:
         self.server.shutdown()
         self.server_thread.join()
+        
+    def setConsoleTitle(self): 
+        ctypes.windll.kernel32.SetConsoleTitleW("VRCHeadpatCounter")
 
     # Entry point from OSC Unity receiver for any contact point
     # Remember from the README that args is derived from your avatars OSC JSON file
     # Per the README these address can only be boolean
     def _def_osc_dispatch(self, address: str, *args) -> None:
         if (self.messenger.timer.enough_time_passed() and \
-                self.messenger.has_new_content(tracker := self.filemanager.get_tracker())):
```

Comment:
Both the indentation you have done here and the one I previously had are valid and shown in the PEP8 style guide.

As personal preference I would like to maintain having the additional indentation as it helps me quickly differentiate between the long if statement and the rest of the code block.